miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java

```diff
@@ -733,11 +734,41 @@ public ModelAndView setupForm(@PathVariable Long libraryId, @PathVariable Long s
     }
   }
 
-  private final BulkTableBackend<Library, LibraryDto, Sample> libraryBulkBackend = new BulkTableBackend<Library, LibraryDto, Sample>(
-      "Libraries", "Samples", "library") {
+  private final BulkEditTableBackend<Library, LibraryDto> libraryBulkEditBackend = new BulkEditTableBackend<Library, LibraryDto>(
+      "library", LibraryDto.class, "Libraries") {
 
     @Override
-    protected LibraryDto packageDtoFromParent(Sample sample) {
+    protected Iterable<Library> load(List<Long> ids) throws IOException {
+      List<Library> results = libraryService.listByIdList(ids);
+      SampleClass sampleClass = null;
+      for (Library library : results) {
+        if (!isDetailedSampleEnabled()) {
+          // Do nothing about sample classes.
+        } else if (sampleClass == null) {
+          sampleClass = ((DetailedSample) library.getSample()).getSampleClass();
+        } else if (((DetailedSample) library.getSample()).getSampleClass().getId() != sampleClass.getId()) {
+          throw new IOException("Can only update libraries when samples all have the same class.");
```

Comment:
Not that I'm aware of. It could cause frustration since the sample class isn't visible on the List Libraries page, and the SampleClass isn't what actually matters. Deal with it later if it becomes a problem I guess. :+1: